src/detect-parse.c

```diff
@@ -858,6 +864,76 @@ int SigMatchListSMBelongsTo(const Signature *s, const SigMatch *key_sm)
     return -1;
 }
 
+/**
+ * \brief Parse and setup a direction
+ *
+ * \param s signature
+ * \param str argument to the keyword
+ * \param only_dir argument wether the keyword only accepts a direction
+ *
+ * \retval 0 on success, -1 on failure
+ */
+static int DetectSetupDirection(Signature *s, char **str, bool only_dir)
```

Comment:
why not a single pointer `char *str`? We're not passing something back, are we?

Reply:
> We're not passing something back, are we?

Yes, we are moving forward the string pointer, so that the keyword `filesize` only "sees" `>100` when the signature had `filesize: to_client, >100;`